lib/OpenQA/CacheService.pm

```diff
@@ -140,3 +141,30 @@ Enqueue the task. It acceps a POST JSON payload of the form:
       {task => 'cache_assets', args => [qw(42 test hdd open.qa)], lock => 'some lock'}
 
 =cut
+
+__DATA__
+@@ cache_service
+-- 1 up
+CREATE TABLE IF NOT EXISTS assets (
+    `etag` TEXT,
+    `size` INTEGER,
+    `last_use` DATETIME NOT NULL,
+    `filename` TEXT NOT NULL UNIQUE,
+    PRIMARY KEY(`filename`)
+);
+
+-- 1 down
+DROP TABLE assets;
+
+-- 2 up
+CREATE TABLE IF NOT EXISTS downloads (
+    `id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    `lock` TEXT,
+    `job_id` TEXT,
+    `created` DATETIME NOT NULL DEFAULT current_timestamp
+);
+CREATE INDEX IF NOT EXISTS downloads_lock on downloads (lock);
+CREATE INDEX IF NOT EXISTS downloads_created on downloads (created);
+
+-- 2 down
+DROP TABLE downloads;
```

Comment:
curious, how did you come up with this `_DATA_` section?

Reply:
What do you mean? It was previously in `OpenQA::CacheService::Model::Cache` for historic reasons, but now the schema spans two classes (`OpenQA::CacheService::Model::Downloads`). So moving it into the application class made sense.

Reply:
I could also have moved the migrations into an external SQL file, similar to how [Minion](https://github.com/mojolicious/minion/blob/master/lib/Minion/Backend/resources/migrations/pg.sql) does it, but we have no naming convention for that in openQA yet and i wasn't too keen on arguing about it. 😉

Reply:
If you're not seen this migration format yet, it's from [Mojo::SQLite::Migrations](https://metacpan.org/pod/Mojo::SQLite::Migrations#DESCRIPTION). Really simple, just one SQL file with comments separating the versions.

Reply:
ok, thx for explanation. As I stated, I was just "curious", not at all critic or something.

Reply:
The question was just a bit vague.